### ISSUE TITLE
Add option <createClasspath> to <build>

### ISSUE THREAD

**Issue by rhuss** [rhuss/docker-maven-plugin#283]
If given within a `<build>` section like in 

```
<build>
  <!-- ... -->
 <assembly>
   <createClasspath>run-classpath</createClasspath>
   ....
    <basedir>/opt/demo</basedir>
    <descriptor>assembly.xml</descriptor>
  </assembly>
</build>
```

the plugin will add a classpath to the given path within the basedir and add it to the Dockerfile. This file should also respect any filename mappings happening in the assembly.xml.

This can be done by re-using the classpath resolution used in the `maven-dependency-plugin` (unfortunately not easy reusable except when subclassing) and the [MappingTrackingArchiver](https://github.com/rhuss/docker-maven-plugin/blob/master/src/main/java/org/jolokia/docker/maven/assembly/MappingTrackArchiver.java) we are using for the watch feature. If this project is of packaging `jar` or `war` then it should be added to the classpath, too.  

The use case is for flat classpath apps which come with a startup script which can use this generated file for setting up the classpath properly.

**Comment by rhuss** [comment]
There is a super easy solution by using the `maven-dependency-plugin`:

```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-dependency-plugin</artifactId>
  <version>2.10</version>
  <executions>
    <execution>
      <id>attach-deps</id>
      <phase>package</phase>
      <goals>
        <goal>build-classpath</goal>
      </goals>
      <configuration>
        <prefix>.</prefix>
        <includeScope>runtime</includeScope>
        <outputFile>${project.build.directory}/classpath</outputFile>
      </configuration>
    </execution>
  </executions>
</plugin>
```

This will create a file `classpath` in `target/` which will be automatically picked up by the `docker-maven-plugin` when using the assembly descriptor `artifactWithDependencies` (for the next version).

**Comment by osigida** [comment]
I've solved it with existing tool set like that
```
                                    <assembly>
                                        <basedir>/</basedir>
                                        <inline>
                                            <dependencySet>
                                                <outputDirectory>jars</outputDirectory>
                                            </dependencySet>
                                        </inline>
                                 </assembly>
```

this will copy all the dependencies to `/jars` folder

then start up script looks like
```
                                    <entryPoint>
                                        <exec>
                                            <arg>java</arg>
                                            <arg>-cp</arg>
                                            <arg>jars/*</arg>
                                            <arg>${mainclass}</arg>
                                        </exec>
                                    </entryPoint>
```

**Comment by rhuss** [comment]
@osigida yeah, works, but leaves you with a random class path, i.e. the order is not defined and that might cause issue for conflicting jars (which, btw, is not so uncommon). The solution [above](https://github.com/rhuss/docker-maven-plugin/issues/283#issuecomment-143771149) will take the dependency ordering (as it it given in the pom.xml) into account by creating a dedicated classpath file.

**Comment by rhuss** [comment]
You could use the same 'trick' if you use a shell exntryPoint which does a `$(cat jars/classpath)` instead of `-cp jars/*` (however you would need the shell).

Or use a Java base image from [fabric8io/base-images](https://github.com/fabric8io/base-images) which provides a sophisticated run script (from [fabric8/run-java-sh](https://github.com/fabric8io/run-java-sh)) which takes this into account (+ you have hooks for remote debugging, installing an Jolokia and/or Prometheus agent, ...)
